ControlSettings/ControlSettingsCommon.h

```diff
@@ -0,0 +1,78 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2022 Sky UK
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef CONTROLSETTINGS_COMMON_H
+#define CONTROLSETTINGS_COMMON_H
+
+
+#define VIDEO_DESCRIPTION_MAX (25)
+#define VIDEO_DESCRIPTION_NAME_SIZE (25)
+
+#define DECLARE_JSON_RPC_METHOD(method) \
+uint32_t method(const JsonObject& parameters, JsonObject& response);
```

Comment:
please develop a JSON RPC interface schema

like it's done for all plugins

https://github.com/rdkcentral/rdkservices#documentation

add it to https://github.com/rdkcentral/ThunderInterfaces/tree/R2/jsonrpc

this will generate you header file with classes for input and output objects

do not hand-craft that in bare Json Object

Reply:
Let me go through the documents and comeback on this.
Thanks.